shared/defs/mapObjectsTyping.ts

```diff
@@ -0,0 +1,454 @@
+            error: string
+        }
+        casingImg?: {
+            sprite: string
+            pos: Vec2
+            scale: number
+            alpha: number
+            tint: number
+        }
+        locked?: boolean
+    }
+    hinge?: Vec2
+    isWindow?: boolean
+    destroyType?: string
+    stonePlated?: boolean
+    aabb?: AABBWithHeight
+    isTree?: boolean
+    button?: {
+        interactionRad: number
+        interactionText: string
+        useOnce: boolean
+        useType?: string
+        useDelay: number
+        useDir: Vec2
+        useImg: string
+        sound: {
+            on: string
+            off: string
+        }
+        destroyOnUse?: boolean
+        useParticle?: string
+        offImg?: string
+    }
+    disableBuildingOccupied?: boolean
+    damageCeiling?: boolean
+    lootSpawn?: {
+        offset: Vec2
+        speedMult: number
+    }
+    dropCollision?: AABBWithHeight
+    airdropCrate?: boolean
+    isBush?: boolean
+    isDecalAnchor?: boolean
+    swapWeaponOnDestroy?: boolean
+    regrow?: boolean
+    regrowTimer?: number
+    armorPlated?: boolean
+    smartLoot?: boolean
+    createSmoke?: boolean
+}
+
+//
+// BuildingDef
+//
+export interface BuildingDef {
+    type: "building"
+    map?: {
+        display?: boolean
+        color?: number
+        scale?: number
+        shapes?: Array<{
+            collider: Collider & { height: number }
+            color: number
+        }>
+        displayType?: string
+    }
+    terrain: {
+        grass?: boolean
+        beach?: boolean
+        lakeCenter?: boolean
+        waterEdge?: {
+            dir: Vec2
+            distMin: number
+            distMax: number
+        }
+        spawnPriority?: number
+        riverShore?: boolean
+        nearbyRiver?: {
+            radMin: number
+            radMax: number
+            facingOri: number
+        }
+        bridge?: {
+            nearbyWidthMult: number
+        }
+    }
+    mapObstacleBounds?: Array<Collider & { height: number }>
+    zIdx?: number
+    floor: {
+        surfaces: Array<{
+            type: string
+            collision: AABBWithHeight[]
+            data?: {
+                isBright: boolean
+            }
+        }>
+        imgs: FloorImage[]
+    }
+    ceiling: {
+        zoomRegions: Array<{
+            zoomIn?: AABBWithHeight
+            zoomOut?: AABBWithHeight
+            zoom?: number
+        }>
+        vision?: {
+            dist?: number
+            width?: number
+            linger?: number
+            fadeRate?: number
+        }
+        imgs: FloorImage[]
+        damage?: {
+            obstacleCount: number
+        }
+        destroy?: {
+            wallCount: number
+            particle: string
+            particleCount: number
+            residue: string
+            sound?: string
+        }
+        collision?: AABBWithHeight[]
+    }
+    mapObjects: Array<{
+        type?: string | (() => string)
```

Comment:
remove `?` from it so you can remove the ts-expect-error on `map.ts`

Reply:
Hey thanks for taking a look but this isn't ready for review yet, and now that I've pushed the client code, it won't be for a while.

#36 is ready tho.